### PR TITLE
feature: add dark-mode

### PR DESCRIPTION
**Pull request by fuwn**

### CODE REVIEW

**Comment by Bios-Marcel** [comment]
Sorry, but as it is I will not merge this.

The darkmode code should only contain what is necessary. I am assuming you've copied this somewhere?
In this case it'd also be interesting as to how this is licensed.

I am sure this for example wasn't added intentionally:

```css
a[href="https://coinmarketcap.com/"] 
```

Either way, why does the file exist twice? Once standalone and once in the base.css?

If I were to do this, I'd do variables at the top of base.css and the other css files which can be used throughout the code without doing one big difference in the media query. However, inverting the black SVGs and PNGs isn't a bad idea, I'd approve of that for now.

**Comment by fuwn** [comment]
> Sorry, but as it is I will not merge this.
> 
> 
> 
> The darkmode code should only contain what is necessary. I am assuming you've copied this somewhere?
> 
> In this case it'd also be interesting as to how this is licensed.
> 
> 
> 
> I am sure this for example wasn't added intentionally:
> 
> 
> 
> ```css
> 
> a[href="https://coinmarketcap.com/"] 
> 
> ```
> 
> 
> 
> Either way, why does the file exist twice? Once standalone and once in the base.css?
> 
> 
> 
> If I were to do this, I'd do variables at the top of base.css and the other css files which can be used throughout the code without doing one big difference in the media query. However, inverting the black SVGs and PNGs isn't a bad idea, I'd approve of that for now.

I used dynamic theme generation from Darkreader, it's generated on the fly.

Your question is answered at the top of the file in question.

**Comment by Bios-Marcel** [comment]
> The "coinmarketcap" reference was likely included because the algorithm found a reference to it somewhere on the site.

Well, not in the codebase at least. Can you prove this somehow, that'd be terrifying :P

> I used dynamic theme generation from Darkreader, it's generated on the fly based on the site its self.

Well, like said, I'd prefer self-crafted code. As it's easier to understand and easier to maintain and also saves the user some bytes ;)

**Comment by fuwn** [comment]
Refactored. Compared to Darkreader's generated copy, this removes a lot of the bloat and repetition which was included.

File: communication/resources/base.css [hunk 1]
```diff
@@ -95,110 +95,110 @@ ul {
...
+        background-color: rgb(76, 68, 53) !important;
+        border-bottom-color: rgb(64, 57, 45) !important;
+    }
+    .tab-content {
+        border-color: initial !important;
+        background-color: rgb(35, 31, 24) !important;
+    }
+    .table-wrapper { border-color: rgb(121, 108, 84) !important; }
+    .lobby-details { background-color: rgb(58, 52, 40) !important; }
+    .vimvixen-hint {
```
Comment (Bios-Marcel) [inline communication/resources/base.css:146]: this is useless too i guess?

Comment (fuwn) [reply communication/resources/base.css:146]: `.vimvixen-hint` possibly, however `.tabble-wrapper` and `.lobby-details`; despite not having previous CSS assignments, these could be a light colour be default so these are just assigning the dark colour to them.

Comment (Bios-Marcel) [reply communication/resources/base.css:146]: yeah, i just meant the vim thing

Comment (fuwn) [reply communication/resources/base.css:146]: Removing it right now.

File: communication/resources/base.css [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+    --selection-text: #ffffd8 !important;
+}
+
 html, body {
     height: 100vh;
 }
 
 body {
-    font-family: 'Montserrat';
+    font-family: 'Montserrat', serif;
```
Comment (fuwn) [inline communication/resources/base.css:13]: Forgot to mention in the commit, but I added a generic font-family fallback here.

Comment (Bios-Marcel) [reply communication/resources/base.css:13]: What for though? The font is shipped with scribble.rs

**Comment by Bios-Marcel** [comment]
Well, i get a syntax error. Either way, I checked it out for the first time and ngl, I hate the colors ^^

**Comment by fuwn** [comment]
> https://scribble-rs.github.io/articles/welcome.html
> 
> Here, just made the blog today. It has datk / lightmode support. I very much like the colors they and wanted to use the same for scribble.rs I thought

I'll re-open and look into mapping the blog's colour scheme either tomorrow or the day after.

File: communication/resources/base.css (hunk 2 above)
Comment (fuwn) [reply communication/resources/base.css:13]: Just general good practice AFAIK.

Also my IDE kept throwing errors to fix it.

**Comment by Bios-Marcel** [comment]
I'll be closing this. As mentioned in the issue, darkmode probably isn't all too useful, since the canvas is white and bright.
This means we'll still burn the users eyes, unless we generally change the canvas color ^^

**Pull request closed without merging**